server/src/main/java/org/opensearch/cluster/coordination/Coordinator.java

```diff
@@ -1330,7 +1333,9 @@ assert getLocalNode().equals(clusterState.getNodes().get(getLocalNode().getId())
                     + clusterState;
 
                 final PublicationTransportHandler.PublicationContext publicationContext = publicationHandler.newPublicationContext(
-                    clusterChangedEvent
+                    clusterChangedEvent,
+                    coordinationState.get().isRemotePublicationEnabled(),
+                    persistedStateRegistry
```

Comment:
why do need to wire them in publication context?

Reply:
`isRemotePublicationEnabled` is used in the `sendPublishRequest` method in PublicationContent to decide whether to remote publish request.
`persistedStateRegistry` gives us access to the `RemotePersistedState` from where we can get the manifest file name which will be used in remote publish request